optimum/utils/normalized_config.py

```diff
@@ -94,6 +94,30 @@ def __getattr__(self, attr_name):
         return super().__getattr__(attr_name)
 
 
+class NormalizedEncoderDecoderConfig(NormalizedSeq2SeqConfig, NormalizedVisionConfig):
+    ENCODER_CONFIG = None
+    DECODER_CONFIG = None
+
+    ENCODER_NORMALIZED_CONFIG_CLASS = None
+    DECODER_NORMALIZED_CONFIG_CLASS = None
+
+    def __getattr__(self, attr_name):
+        if (
+            self.ENCODER_CONFIG is not None
+            and self.ENCODER_NORMALIZED_CONFIG_CLASS is not None
+            and attr_name.upper() in dir(self.ENCODER_NORMALIZED_CONFIG_CLASS)
+        ):
+            return self.ENCODER_NORMALIZED_CONFIG_CLASS.__getattr__(attr_name)
+        if (
+            self.DECODER_CONFIG is not None
+            and self.DECODER_NORMALIZED_CONFIG_CLASS is not None
+            and attr_name.upper() in dir(self.DECODER_NORMALIZED_CONFIG_CLASS)
+        ):
+            return self.DECODER_NORMALIZED_CONFIG_CLASS.__getattr__(attr_name)
+
+        return super().__getattr__(attr_name)
```

Comment:
why do we need this?

Reply:
These type of models can have any decoder and encoder head and it is difficult to manage every normalised class. So created this helper class to work with these models